.circleci/config.yml

```diff
@@ -1,76 +1,69 @@
+    executor: aws-ecr/default
+    steps:
+      - checkout
+      - aws-ecr/build-and-push-image:
+          attach-workspace: true
+          repo: << pipeline.parameters.ecr_repo_name >>
+          dockerfile: "Docker/Dockerfile"
+          tag: "latest,${CIRCLE_BRANCH}-branch,${CIRCLE_SHA1:0:7}-SHA1"
+
+executors:
+  ubuntu:
+    working_directory: ~/tmp_checkout
+    machine:
+      image: ubuntu-2204:2023.04.2
 
 workflows:
-  version: 2
-  build-and-deploy:
+  do:
     jobs:
-      - trafficinfo/build:
+      - gradle/test:
+          name: "Build and test"
+          executor: ubuntu
           context:
-            - trafficinfo
+            - ruteplan
             - nexus
-          jdk_version: "17"
-          post-steps:
-            - trafficinfo/slack-failure
-            - run: ./gradlew -q javaToolchains
-
-      - stepfunction-ci/build-docker:
-          context: trafficinfo
-          enable_buildkit: true
-          ecr_repo_name: << pipeline.parameters.ecr_repo_name >>
-          requires:
-            - trafficinfo/build
+          deps_checksum_file: build.gradle.kts
+          test_command: "clean test build --no-daemon"
           post-steps:
-            - trafficinfo/slack-failure
+            - run:
+                command: ./gradlew -q javaToolchains
+                when: on_fail
+            - run: ls -laR build/docker
+            - persist_to_workspace:
+                root: .
+                paths:
+                  - "build/*"
 
-      - stepfunction-ci/build-repo-zip:
-          context: trafficinfo
-          s3_bucket_name: << pipeline.parameters.s3_bucket_name >>
+      - build-and-push-image:
+          context: ruteplan
           requires:
-            - trafficinfo/build
+            - "Build and test"
 
-          post-steps:
-            - trafficinfo/slack-failure
-
-      - stepfunction-ci/upload-artifacts:
-          context: trafficinfo
-          s3_bucket_name: << pipeline.parameters.s3_bucket_name >>
-          requires:
-            - stepfunction-ci/build-repo-zip
-          post-steps:
-            - trafficinfo/slack-failure
-
-      - stepfunction-ci/upload-trigger-event:
-          context: trafficinfo
-          s3_bucket_name: << pipeline.parameters.s3_bucket_name >>
-          pipeline_name: << pipeline.parameters.pipeline_name >>
-          deployment_repo: << pipeline.parameters.deployment_repo >>
-          requires:
-            - stepfunction-ci/build-docker
-            - stepfunction-ci/upload-artifacts
-          post-steps:
-            - trafficinfo/slack-failure
-
-      - trafficinfo/slack-notify-success:
-          context: trafficinfo
-          message: ":sunglasses: :thumbsup: Uploaded pipeline trigger event for $CIRCLE_PROJECT_REPONAME version $CIRCLE_SHA1."
-          requires:
-            - stepfunction-ci/upload-trigger-event
\ No newline at end of file
+# TODO waiting for infrastructure team to implement repo override for trigger-deployment in orb.
```

Comment:
Denne todo-en skjønner jeg ikke helt.

Reply:
de bruker repo navnet for å trigge pipeline, og ingen mulighet til å override.
I dette tilfellet i baseline, ligger terraform og microservice i to ulike repo, derfor vil det opprettes to egne pipelines hvis man kjører trigger-deployment fra circleci step. Det vil ikke funke, løsningen blir å legge til et parameter i trigger-deployment hvor man kan override deployment pipeline navnet slik at riktig pipeline blir brukt og ny ikke blir opprettet.

Reply:
men det finnes ikke enda, så de må implementere det